src/legacy/core_plugins/kbn_vislib_vis_types/public/editors/gauge.tsx

```diff
@@ -0,0 +1,267 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { EuiButtonEmpty, EuiPanel, EuiSpacer, EuiTitle } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { VisOptionsProps } from 'ui/vis/editors/default';
+import { RangesParamEditor } from 'ui/agg_types/controls/ranges';
+import { SelectOption } from '../controls/select';
+import { SwitchOption } from '../controls/switch';
+import { TextInputOption } from '../controls/text_input';
+import { GaugeVisParams } from '../gauge';
+
+function GaugeOptions(props: VisOptionsProps<GaugeVisParams>) {
+  const { stateParams, setValue, vis, uiState } = props;
+  const [isCustomColors, setIsCustomColors] = useState(false);
+
+  useEffect(() => {
+    uiState.on('colorChanged', () => {
+      setIsCustomColors(true);
+    });
+  }, []);
+
+  const setGaugeValue = useCallback(
+    <T extends keyof GaugeVisParams['gauge']>(paramName: T, value: GaugeVisParams['gauge'][T]) =>
+      setValue('gauge', {
+        ...stateParams.gauge,
+        [paramName]: value,
+      }),
+    [stateParams.gauge]
+  );
+
+  const setGaugeType = useCallback(
+    (paramName: 'gaugeType', value: GaugeVisParams['gauge']['gaugeType']) => {
+      const minAngle = value === 'Arc' ? undefined : 0;
+      const maxAngle = value === 'Arc' ? undefined : 2 * Math.PI;
+
+      setValue('gauge', {
+        ...stateParams.gauge,
+        [paramName]: value,
+        minAngle,
+        maxAngle,
+      });
+    },
+    [stateParams.gauge]
+  );
+
+  const resetColorsButton = (
+    <EuiButtonEmpty
```

Comment:
Looks like the size of this one is not small enough. It might need to be a `EuiLink` inside `EuiText size="xs"`